Assets/Scripts/SlipTime/SlipTimeManager.cs

```diff
@@ -0,0 +1,68 @@
+﻿using UnityEngine;
+
+namespace SlipTime
+{
+    /// <summary>
+    /// Manages activating and deactivating SlipTime.
+    /// </summary>
+    public class SlipTimeManager : MonoBehaviour
+    {
+        /// <summary>
+        /// The percentage by which to scale time in SlipTime.
+        /// </summary>
+        public float slipTimeScalar = 0.25f;
+        
+        /// <summary>
+        /// The default duration of SlipTime in seconds.
+        /// </summary>
+        public float slipTimeDuration = 3f;
+        
+        private bool inSlipTime;
+        private SlipTimeAdherent[] slipTimeAdherents;
+        private double timeCounter;
+
+        // Update is called once per frame
+        private void Start()
+        {
+            slipTimeAdherents = FindObjectsOfType<SlipTimeAdherent>();
```

Comment:
I think Unity probably just allocates an array address for this array so that is why we can just call it once and work through out the program. Since the API doesn't guarantee the memory to be the same, I think it is safer to call it every loop as the example shows: https://docs.unity3d.com/ScriptReference/Object.FindObjectsOfType.html

Reply:
Fixed.